scripts/maint/rename_c_identifier.py

```diff
@@ -0,0 +1,214 @@
+        stdout=subprocess.PIPE,
+        encoding="utf-8")
+    for line in proc.stdout.readlines():
+        line = line.strip()
+        if is_c_file(line):
+            yield line
+
+class Rewriter:
+    """
+       A rewriter applies a series of word-by-word replacements, in
+       sequence.  Replacements only happen at word boundaries.
+
+       >>> R = Rewriter([("magic", "secret"), ("words", "codes")])
+       >>> R.apply("The magic words are rambunctious bluejay")
+       'The secret codes are rambunctious bluejay'
+       >>> R.apply("The magical words are rambunctious bluejay")
+       'The magical codes are rambunctious bluejay'
+       >>> R.get_count()
+       3
+    """
+    def __init__(self, replacements):
+        """Make a new Rewriter. Takes a sequence of pairs of
+           (from_id, to_id), where from_id is an identifier to replace,
+           and to_id is its replacement.
+        """
+        self._patterns = [ ]
+        for id1,id2 in replacements:
+            pat = re.compile(r"\b{}\b".format(re.escape(id1)))
+            self._patterns.append((pat, id2))
+
+        self._count = 0
+
+    def apply(self, line):
+        """Return `line` as transformed by this rewriter."""
+        for pat, ident in self._patterns:
+            line,count = pat.subn(ident, line)
+            self._count += count
+        return line
+
+    def get_count(self):
+        """Return the number of identifiers that this rewriter has
+           rewritten."""
+        return self._count
+
+def rewrite_files(files, rewriter):
+    """
+       Apply `rewriter` to every file in `files`, replacing those files
+       with their rewritten contents.
+    """
+    for line in fileinput.input(files, inplace=True):
+        sys.stdout.write(rewriter.apply(line))
+
+def make_commit_msg(pairs, no_verify):
+    """Return a commit message to explain what was replaced by the provided
+       arguments.
+    """
+    script = [ "./scripts/maint/rename_c_identifier.py" ]
+    for id1, id2 in pairs:
+        qid1 = shlex.quote(id1)
+        qid2 = shlex.quote(id2)
+        script.append("        {} {}".format(qid1,qid2))
+    script = " \\\n".join(script)
+
+    if len(pairs) == 1:
+        line1 = "Rename {} to {}".format(pairs[0])
+    else:
+        line1 = "Replace several C identifiers."
+
+    msg = """\
+{}
+
+This is an automated commit, generated by this command:
+
+{}
+""".format(line1, script)
+
+    if no_verify:
+        msg += """
+It was generated with --no-verify, so it probably breaks some commit hooks.
+The commiter should be sure to fix them up in a subsequent commit.
+"""
+
+    return msg
+
+def commit(pairs, no_verify=False):
+    """Try to commit the current git state, generating the commit message as
+       appropriate.  If `no_verify` is True, pass the --no-verify argument to
+       git commit.
+    """
+    args = []
+    if no_verify:
+        args.append("--no-verify")
+
+    fname = None
+    try:
```

Comment:
I feel like this `try` is unneeded and that putting everything that uses `fname` and `f` inside the `with` block is enough. Is the extra `try` block and the `delete=false` flag addressing these docs? If so, I think that explains the extra wrapping.

> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later)

https://docs.python.org/3.5/library/tempfile.html#tempfile.NamedTemporaryFile

Reply:
7ad0f4b8f9c922d7685fba87c038a63a250a153c